### PR TITLE
Update ContainerDetails.jsx

### PR DESCRIPTION
**Pull request by hurenkam**
Add Network, Volumes and Environment sections to Detailed view

### CODE REVIEW

File: src/ContainerDetails.jsx [hunk 1]
```diff
@@ -74,19 +130,27 @@ const ContainerDetails = ({ container, containerDetail }) => {
...
-            <DescriptionList className='container-details-state'>
+            <DescriptionList className='container-details-networks'>
                 <DescriptionListGroup>
-                    <DescriptionListTerm>{_("Created")}</DescriptionListTerm>
-                    <DescriptionListDescription>{utils.localize_time(Date.parse(container.Created) / 1000)}</DescriptionListDescription>
+                    <DescriptionListTerm>{_("Networks")}</DescriptionListTerm>
+                    <DescriptionListDescription>{networks}</DescriptionListDescription>
                 </DescriptionListGroup>
+            </DescriptionList>
+            <DescriptionList className='container-details-volumes'>
```
Comment (jelly) [inline src/ContainerDetails.jsx:144]: Should not shown when volumes are empty.

**Review by jelly — CHANGES_REQUESTED** [review]

**Comment by jelly** [comment]
Thanks for the PR! This changes the detail view quite a bit, so I've tagged our design for a second opinion @garrett

image

I think we should not show Volumes when they are "empty", I'm not sure about showing the Networks by default as it seems a bit redundant for normal configurations.

**Comment by garrett** [comment]
At first glance, the organization doesn't make much sense, but I guess it does:

Going from left to right:

Basic container details, Networking, Networking (again; this should probably be grouped, right?), Volumes, Environment.

If a section doesn't apply, don't show it.

Also, I guess this is using flex. It probably doesn't wrap properly in smaller sizes. I do have a fix locally for that, which I need to push up and also add to Cockpit's workarounds. (It looks like Podman doesn't import the workarounds?)

File: src/client.js [hunk 2]
```diff
@@ -66,6 +66,28 @@ export function getContainers(system, id) {
...
+    return new Promise((resolve, reject) => {
+        const options = { };
+        if (id)
+            options.filters = JSON.stringify({ id: [id] });
+
+        podmanCall("libpod/networks/json", "GET", options, system)
+                .then(reply => resolve(JSON.parse(reply)))
+                .catch(reject);
+    });
+}
```
[the remark this answers is not in the document]
Comment (hurenkam) [reply src/client.js:89]: Yeah, this was a premature commit, I'm indeed experimenting with global network/volumes sections.

File: src/ContainerDetails.jsx (hunk 1 above)
Comment (hurenkam) [reply src/ContainerDetails.jsx:144]: agreed

**Comment by garrett** [comment]
BTW: The flex fix I talked about landed as a workaround, so things show up more properly across different sizes. It's "just" layout related... so hopefully the layout works as expected now, even in smaller sizes, like in mobile. (This also affected main, not just this PR. I'm only mentioning it here as I mentioned it above too. So it has been resolved, in other words.)

**Comment by garrett** [comment]
I don't know if this is ready for my review yet.

Meanwhile, I tried building it locally as-is and I get this...


Toolbox container is huge with mounts:

Screenshot 2022-03-15 at 09-44-41 Podman containers - garrett@Bolt

But a default Nextcloud image is even quite a bit more with envirnomental variables, and it's not a great layout and ther'es no space between env and created:

Screenshot from 2022-03-15 09-48-33

Here's what Nextcloud looks like on my home server (not using this PR, of course), for reference:

Screenshot 2022-03-15 at 09-48-41 Podman containers - garrett@Snow

**Comment by jelly** [comment]
Testing this PR again with a nextcloud container:

image

image

When there are no volumes, we should not shown the `volumes` header.

**Comment by garrett** [comment]
@marusak, @jelly, and I talked about this PR today and came to these conclusions:

- Mirror the create dialog by moving ports, volumes, and environment into a system integration tab
- Deduplicate the environment variables by comparing with image's environment variables and a few that podman always sets (HOME, etc.) and show the remainder
- Don't show networks _yet_, until we can edit it in Cockpit (there's nothing interesting and actionable yet)

Having this information on a separate tab gives the information more space for layout and makes it more consistent with the create dialog.

**Review by garrett — CHANGES_REQUESTED** [review]
@marusak, @jelly, and I talked about this PR today and came to these conclusions:

- Mirror the create dialog by moving ports, volumes, and environment into a system integration tab
- Deduplicate the environment variables by comparing with image's environment variables and a few that podman always sets (HOME, etc.) and show the remainder
- Don't show networks _yet_, until we can edit it in Cockpit (there's nothing interesting and actionable yet)

Having this information on a separate tab gives the information more space for layout and makes it more consistent with the create dialog.

**Comment by marusak** [comment]
@hurenkam are you still interested in this work and if so, do you follow on what changes need to be made?

**Comment by hurenkam** [comment]
> @hurenkam are you still interested in this work and if so, do you follow on what changes need to be made?

I think I can follow more or less what is meant here, but frankly, I'm not sure I'm the right person to move things around and fix user interface related details. I am familiar with using javascript, json and rest interfaces, and as such can quickly find and expose info that I'm interested in, but am really not familiar with the ui library used in this component, and find it hard to fix layout issues.

The main reason why I have not followed up yet though, is the network info. All of my VM's are using a custom network, some even have several custom networks. To expose that properly, there should be multiple Gateway/Mac/IP/Subnet entries. That requires some rethinking on how to expose that information properly. I'm guessing you came to a similar conclusion which is why you're suggesting to postpone this part?

Going forward, I'm willing to give this another attempt, provided you guys are not in a hurry.

**Comment by tomasmatus** [comment]
Hi @hurenkam! Yesterday I started working on adding volumes and env variables. Following Garrett's comment I believe UI changes are finished but I still need to add tests before I can open PR.

**Comment by garrett** [comment]
> Following Garrett's comment I believe UI changes are finished but I still need to add tests before I can open PR.

I'm looking forward to the changes when you share them (either in this PR or in a separate one)! Please make sure to add me as a reviewer (if it's a separate PR) and ping me on IRC with the link when it's available.

**Pull request closed without merging**